### ISSUE TITLE
Chosen interface has no ip address caused kube-proxy failed to start

### ISSUE THREAD

**Issue by ddysher** [GoogleCloudPlatform/kubernetes#3336]
kube-proxy finds the first up interface (non p2p, loopback) as its host interface, but that interface doesn't have an ip address on my machine.

https://github.com/GoogleCloudPlatform/kubernetes/blob/53a9d106c4aabcd550cc32ae4e8004f32fb0ae7b/pkg/proxy/proxier.go#L760

Specifically, kube-proxy finds eth0, which doesn't have ip address associated with it (wlan0 does).  I don't know why eth0 no longer gets an ip address from dhcp server, it worked before.

**Comment by ddysher** [comment]
A reference fix #3337

**Comment by ddysher** [comment]
@thockin Here is some details
```
$ ifconfig:
docker0   Link encap:Ethernet  HWaddr 56:84:7a:fe:97:99  
          inet addr:172.17.42.1  Bcast:0.0.0.0  Mask:255.255.0.0
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

eth0      Link encap:Ethernet  HWaddr 74:d4:35:18:b9:75  
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:14295 errors:0 dropped:0 overruns:0 frame:0
          TX packets:14295 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:1417795 (1.4 MB)  TX bytes:1417795 (1.4 MB)

wlan0     Link encap:Ethernet  HWaddr 10:fe:ed:98:7a:a0  
          inet addr:10.0.0.3  Bcast:10.0.0.255  Mask:255.255.255.0
          inet6 addr: fe80::12fe:edff:fe98:7aa0/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:2047278 errors:0 dropped:0 overruns:0 frame:0
          TX packets:3176068 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:509796709 (509.7 MB)  TX bytes:3272495534 (3.2 GB)
```

Run local-up:
```
I0108 20:59:53.924829   20589 proxier.go:775] Choosing interface eth0 for from-host portals
panic: runtime error: index out of range

goroutine 16 [running]:
runtime.panic(0x85a360, 0xb41d1c)
	/usr/local/go/src/pkg/runtime/panic.c:279 +0xf5
github.com/GoogleCloudPlatform/kubernetes/pkg/proxy.chooseHostInterface(0x0, 0x0, 0x0, 0x0, 0x0)
	/home/deyuan/code/source/go-workspace/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/proxy/proxier.go:780 +0x5cb
github.com/GoogleCloudPlatform/kubernetes/pkg/proxy.NewProxier(0x7fabefe31cd0, 0xc20807efc0, 0xc2080a7200, 0x10, 0x10, 0x7fabefe31c80, 0xc208045e00, 0xc208077df8)
	/home/deyuan/code/source/go-workspace/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/proxy/proxier.go:319 +0xed
main.main()
	/home/deyuan/code/source/go-workspace/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/cmd/kube-proxy/proxy.go:73 +0x3c8

goroutine 17 [runnable]:
runtime.MHeap_Scavenger()
	/usr/local/go/src/pkg/runtime/mheap.c:507
runtime.goexit()
	/usr/local/go/src/pkg/runtime/proc.c:1445

goroutine 18 [runnable]:
bgsweep()
	/usr/local/go/src/pkg/runtime/mgc0.c:1976
runtime.goexit()
	/usr/local/go/src/pkg/runtime/proc.c:1445

goroutine 19 [runnable]:
runfinq()
	/usr/local/go/src/pkg/runtime/mgc0.c:2606
runtime.goexit()
	/usr/local/go/src/pkg/runtime/proc.c:1445

goroutine 20 [runnable]:
github.com/golang/glog.(*loggingT).flushDaemon(0xb48460)
	/home/deyuan/code/source/go-workspace/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:878
created by github.com/golang/glog.init·1
	/home/deyuan/code/source/go-workspace/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:410 +0x2b2

goroutine 17 [syscall]:
runtime.goexit()
	/usr/local/go/src/pkg/runtime/proc.c:1445

goroutine 21 [runnable]:
github.com/GoogleCloudPlatform/kubernetes/pkg/util.Forever(0x9a4f80, 0x12a05f200)
	/home/deyuan/code/source/go-workspace/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/util/util.go:56
created by github.com/GoogleCloudPlatform/kubernetes/pkg/util.InitLogs
	/home/deyuan/code/source/go-workspace/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/util/logs.go:48 +0x9a

goroutine 22 [runnable]:
github.com/GoogleCloudPlatform/kubernetes/pkg/proxy/config.watchForUpdates(0xc20807eea0, 0x7fabefe31be0, 0xc20807ede0, 0xc208004e40)
	/home/deyuan/code/source/go-workspace/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/proxy/config/config.go:264
created by github.com/GoogleCloudPlatform/kubernetes/pkg/proxy/config.NewServiceConfig
	/home/deyuan/code/source/go-workspace/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/proxy/config/config.go:182 +0x1b3

goroutine 23 [runnable]:
github.com/GoogleCloudPlatform/kubernetes/pkg/proxy/config.watchForUpdates(0xc20807ef90, 0x7fabefe31c30, 0xc20807eed0, 0xc208004ea0)
	/home/deyuan/code/source/go-workspace/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/proxy/config/config.go:264
created by github.com/GoogleCloudPlatform/kubernetes/pkg/proxy/config.NewEndpointsConfig
	/home/deyuan/code/source/go-workspace/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/proxy/config/config.go:87 +0x1b3
```

**Comment by thockin** [comment]
Unclear: are you asking for debug help or just posting status? :)

**Comment by ddysher** [comment]
Both :)   The posted texts are output from my machine.  The fix in #3337 is adding checks to make sure 'addrs' is not out of range, which is the case locally.

**Comment by ddysher** [comment]
I've updated #3337 

Looks like the problem is with Go

**Comment by ddysher** [comment]
Fixed by #3469
